glslc/src/file_compiler.cc

```diff
@@ -77,7 +91,29 @@ bool FileCompiler::CompileShaderFile(const std::string& input_file,
                 << output_file_name_ << "'" << std::endl;
     }
   }
-  return compilation_success;
+
+  // Handle the error message for failing to deduce the shader kind.
+  if (result.GetCompilationStatus() ==
+      shaderc_compilation_status_invalid_stage) {
+    if (IsGlslFile(error_file_name)) {
+      std::cerr << "glslc: error: "
+                << "'" << error_file_name << "': "
+                << ".glsl file encountered but no -fshader-stage "
+                   "specified ahead";
+    } else if (error_file_name == "<stdin>") {
+      std::cerr
+          << "glslc: error: '-': -fshader-stage required when input is from "
+             "standard "
+             "input \"-\"";
+    } else {
+      std::cerr << "glslc: error: "
+                << "'" << error_file_name << "': "
+                << "file not recognized: File format not recognized";
```

Comment:
The error messages about 'invalid shader kind/stage' are moved to here.
Note that, at this point, we already transform '-' to '<stdin>' when the input is from standard input. The error message here is to meet the demand of current tests.